src/gui/widgets/multiline_text.hpp

```diff
@@ -132,8 +132,14 @@ class multiline_text : public text_box_base
 		update_layout();
 	}
 
-	/** Inherited from text_box_base. */
-	void set_cursor(const std::size_t offset, const bool select, const bool autoscroll = true)
+	/** Inherited from text_box_base, defaults the autoscroll argument to true. */
+	void set_cursor(const std::size_t offset, const bool select) override
+	{
+		// Because the two-argument form is virtual, adding a three-argument form requires
+		// a separate function, otherwise callers in text_box_base will ignore the override.
+		set_cursor(offset, select, true);
+	}
+	virtual void set_cursor(const std::size_t offset, const bool select, const bool autoscroll)
```

Comment:
I rewrote the scrolling code in the last few commits, and it seems that the autoscroll check is not needed anymore. So you can remove the that check altogether and make it 2 argument. (also remove the corresponding method call from here : https://github.com/wesnoth/wesnoth/blob/8d3a52eb3fb4e32c5e013eac0bf0a5d7c63839ec/src/gui/widgets/multiline_text.cpp#L215) and remove the comment as well.

Reply:
To test it, just paste some big text and try to navigate and type something, just like in any text editor.

Reply:
My preference in that case would be to close this PR, and wait for one from you.